From 269b2808f1c1d863fe49c96890f922945e609f98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mike=20F=C3=A4hrmann?= <mike_faehrmann@web.de>
Date: Thu, 8 Aug 2024 09:34:51 +0200
Subject: [PATCH] [deviantart] fix "KeyError - 'category'" (#5960, #5961)

'deviation' objects returned by the OAuth API no longer include a
'category' or 'category_path' value.
---
 gallery_dl/extractor/deviantart.py | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/gallery_dl/extractor/deviantart.py b/gallery_dl/extractor/deviantart.py
index a70710c823..f3ea4e70ce 100644
--- a/gallery_dl/extractor/deviantart.py
+++ b/gallery_dl/extractor/deviantart.py
@@ -12,7 +12,6 @@
 from .. import text, util, exception
 from ..cache import cache, memcache
 import collections
-import itertools
 import mimetypes
 import binascii
 import time
@@ -246,7 +245,6 @@ def prepare(self, deviation):
             deviation["username"] = deviation["author"]["username"]
             deviation["_username"] = deviation["username"].lower()
 
-        deviation["da_category"] = deviation["category"]
         deviation["published_time"] = text.parse_int(
             deviation["published_time"])
         deviation["date"] = text.parse_timestamp(
@@ -301,15 +299,6 @@ def _commit_journal_html(self, deviation, journal):
             )
         else:
             needle = '<div usr class="gr">'
-            catlist = deviation["category_path"].split("/")
-            categories = " / ".join(
-                ('<span class="crumb"><a href="{}/{}/"><span>{}</span></a>'
-                 '</span>').format(self.root, cpath, cat.capitalize())
-                for cat, cpath in zip(
-                    catlist,
-                    itertools.accumulate(catlist, lambda t, c: t + "/" + c)
-                )
-            )
             username = deviation["author"]["username"]
             urlname = deviation.get("username") or username.lower()
             header = HEADER_TEMPLATE.format(
@@ -318,7 +307,6 @@ def _commit_journal_html(self, deviation, journal):
                 userurl="{}/{}/".format(self.root, urlname),
                 username=username,
                 date=deviation["date"],
-                categories=categories,
             )
 
         if needle in html:
@@ -624,7 +612,7 @@ def deviations(self):
     def _make_deviation(self, url, user, index, fmt):
         return {
             "author"         : user,
-            "category"       : "avatar",
+            "da_category"    : "avatar",
             "index"          : text.parse_int(index),
             "is_deleted"     : False,
             "is_downloadable": False,
@@ -1773,9 +1761,6 @@ def base36_from_id(deviation_id):
 <span class="user-symbol regular"></span></span></span>,
             <span>{date}</span>
         </li>
-        <li class="category">
-            {categories}
-        </li>
     </ul>
 </div>
 """
